site/src/_pages/spec/1_block-types.mdx

```diff
@@ -37,33 +46,60 @@ A block package SHOULD contain:
     - `image` – a preview image of the block for users to see it in action before using it. This would ideally have a 3:2 width:height ratio and be a minimum of 900x1170px.
 
 ## Data transfer
 
 ### Data provided to blocks
 
 Blocks MUST use the data properties specified in their schema, if any.
-They can expect these properties to be made available to them by the embedding application, exactly how depending on [rendering context](https://blockprotocol.org/spec/implementation-approaches).
-For example, for a React component block they would be passed as ‘props’.
-In a block with an HTML entrypoint, appropriately sandboxed by the embedding application, the properties would be in the global scope (of the sandbox).
 
-The data for the block itself will be at the root level of the data made available to it, i.e. a block which had a `level` property at the root of the properties in its schema would have a `level` property passed in as a prop (or attached to the global scope of sandboxed HTML blocks).
+They can expect these properties to be made available to them by the embedding application, exactly how depending on [rendering context](https://blockprotocol.org/spec/implementation-and-rendering-context).
+
+The data for the block itself will be at the root level of the data made available to it, i.e. a block which had a `level` property at the root of the properties in its schema would have a `level` property passed to it.
+
+#### Fields identifying entities
+
+The block’s data represents an ‘entity’ in the system. Alongside the properties declared in its schema, blocks MUST expect certain fields to be provided by the embedding application, and MUST pass these fields back when requesting action with specific entities:
+
+- `entityId`: an identifier for the entity
+- `entityTypeId`: an identifier for the type/class of the entity
+- `accountId`: an identifier the account/namespace/user to which the entity belongs
+
+These fields MUST be strings. Some MAY be left undefined.
+
+The exact form of these will differ across applications – e.g. some might use human-readable strings, others might use integers (passed as strings) or uuids.
+
+<Frame emoji="🤔">
+
+It's vital to have a way of blocks identifying the entities they target for retrieval or modification.
+
+The approach described above attempts to identify a selection of fields which in combination are sufficient to identify an entity for any application, with applications left to decide which to define.
+
+Ideally, blocks would be able to rely on the concatenation of any supplied identifiers as a unique identifier for the entity within that application, so that the block can track unique entities. An alternative is to require in the specification that `entityId` is unique in the application, and leave it up to embedding applications to make it so.
```

Comment:
If we end up adopting this approach we'll want to replace "Ideally" with "MUST" or "SHOULD" so blocks can reliably depend upon concatenation as described.

Reply:
Agreed